### DISCUSSION TITLE
GLIBC linking issues

### DISCUSSION THREAD

**Discussion by Raj2032** [cross-rs/cross#1531] [Q&A]
On Windows I installed `cross` and `docker` and managed to compile my project just fine using this command:

```cmd
cross build --target i686-unknown-linux-gnu --release
```

However when running this on a 32bit linux operating system I get the following error

```cmd
./updatesql_email_err
./updatesql_email_err: /lib/libc.so.6: version `GLIBC_2.29' not found (required by ./updatesql_email_err)
./updatesql_email_err: /lib/libc.so.6: version `GLIBC_2.9' not found (required by ./updatesql_email_err)
./updatesql_email_err: /lib/libc.so.6: version `GLIBC_2.16' not found (required by ./updatesql_email_err)
./updatesql_email_err: /lib/libc.so.6: version `GLIBC_2.15' not found (required by ./updatesql_email_err)
./updatesql_email_err: /lib/libc.so.6: version `GLIBC_2.28' not found (required by ./updatesql_email_err)
./updatesql_email_err: /lib/libc.so.6: version `GLIBC_2.18' not found (required by ./updatesql_email_err)
```

I want to link all this inside the binary itself.

**Comment by andber1** [comment]
> I want to link all this inside the binary itself.

So you probably want to use the `i686-unknown-linux-musl` target. But it seems that it is not supported by cross-rs. You could build a new image for it. But this target seems to have some problems in general. See [here](https://github.com/rust-lang/rustup/issues/2984).

Another option is to set the glibc version, so that it is compatible with your target system. See [here](https://github.com/cross-rs/cross/discussions/1099) for more details.
